src/polyfill/objassign.js

```diff
@@ -0,0 +1,28 @@
+// Object.assig polyfile from https://github.com/google/closure-compiler/blob/master/src/com/google/javascript/jscomp/js/es6/util/assign.js
+
+/**
+ * Equivalent to the Object.assign() method, but guaranteed to be available for use in code
+ * generated by the compiler.
+ *
+ * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign
+ *
+ * Copies values of all enumerable own properties from one or more
+ * sources to the given target object, and returns the target.
+ *
+ * @final
+ * @param {!Object} target The target object onto which to copy.
+ * @param {...?Object} var_args The source objects.
+ * @return {!Object} The target object is returned.
+ */
+if (Object.assign === 'undefined') {
+  Object.assign = function(target, source) {
```

Comment:
(When there is just one use then `var objAssign = Object.assign` does add size.)

Reply:
Regarding the code size issue,  I agree that we should consider adding our own short names if/when it makes sense, but we should not single out `Object.assign` for this treatment.  For example we could say exacatly the same thing about other global names such as `WebAssembly.instantiateStreaming`.

Regarding polyfilling, I agree we should come up with policy.   At the very least the user should be able to disable polyfils if they wish.   I will open an issue discussion.

Reply:
Polyfills can be added by other build tools. IMO if Emscripten adds any itself it should be opt-in, not opt-out.

Reply:
I'm tempted to go with opt-in so things just work by default... but I could be persuaded otherwise.    We have historically tended to assume that our output should "just work".   We can followup on #15938.

Reply:
We also have a historical precedent here that we have been poly-filling `Promise` on older platforms for while now and nobody has asked us about an opt out (possibly because very few users are targeting these older platforms).   Another option: Force users to opt in or out explicitly when older browser is targeted.

Reply:
Are polyfills only added if you manually set old browsers as supported? That basically is opting in.

Reply:
Exactly, yes.  By default I don't think we would want any polyfills in the output (and we don't have any today).